src/components/Menu/Menu.tsx

```diff
@@ -0,0 +1,288 @@
+  className?: string;
+  /** Optional ID string that other components can cross reference for accessibility purposes. */
+  id?: string;
+  /** Optional string used to identify and highlight an item when the menu opens. The value should
+   * match the id associated with one of the items. */
+  selectedItem?: string;
+  /** Used to set the highlight color for the current item.  The values correspond with the NYPL section colors */
+  highlightColor?: SectionTypes;
+  /** Required string used to set the label text for the button element. If showLabel is false,
+   * this value is instead used to set an aria-label attribute on the button.  The labelText prop is
+   * required for accessibility compliance. */
+  labelText: string;
+  /** Optional string used to set the placement of the list element. */
+  listAlignment?: "left" | "right";
+  /** Required array of data objects used to populate the items rendered in the list.
+   * The order of the array elements will determine the order of the items in the list.*/
+  listItemsData: ListItemsData[];
+  /** Optional boolean value used to toggle the visibility of a border around the button element.*/
+  showBorder?: boolean;
+  /** Optional boolean value used to toggle the visibility of the label text for the button element.
+   * If false, this value will be used to set an aria-label attribute on the button element.  */
+  showLabel?: boolean;
+}
+
+/**Type for the icons/images displayed inline with menu items. */
+interface Media {
+  type: string;
+  name?: IconNames;
+  src?: string;
+  alt?: string;
+}
+
+/** The three types of menu items: */
+export interface ActionItem {
+  type: "action";
+  label: string;
+  id: string;
+  media?: Media | null;
+  onClick: (id: string) => void;
+}
+
+export interface GroupItem {
+  type: "group";
+  id: string;
+  label: string;
+  children: ListItemsData[];
+}
+
+export interface DividerItem {
+  type: "divider";
+  id: string;
+}
+
+/** Menu items can be one of the three types: */
+export type ListItemsData = ActionItem | GroupItem | DividerItem;
+
+export const Menu = chakra(
+  forwardRef<HTMLDivElement, MenuProps>(
+    (
+      {
+        className,
+        id,
+        labelText,
+        listAlignment = "left",
+        selectedItem,
+        highlightColor = "blogs",
+        showBorder = true,
+        showLabel = true,
+        listItemsData,
+        ...rest
+      },
+      ref?
+    ) => {
+      const styles = useMultiStyleConfig("Menu", {
+        highlightColor,
+        showBorder,
+        showLabel,
+      });
+
+      /**  Handle selection alongside custom onClick behavior. */
+      const [selected, setSelected] = useState(selectedItem);
+      const handleSelect = (id, customHandler) => {
+        if (customHandler) {
+          customHandler(id);
+        }
+        setSelected(id);
+      };
+
+      /**  Check props. */
+      if (!listItemsData) {
+        console.warn(
```

Comment:
I missed this earlier but can you add tests to verify that these messages are being logged? Here's an example https://github.com/NYPL/nypl-design-system/blob/development/src/components/Checkbox/Checkbox.test.tsx#L232

For this specific message (and perhaps others), typescript will complain about required props but you can get around that by adding `// @ts-ignore` directly above. For example: https://github.com/NYPL/nypl-design-system/blob/development/src/components/Checkbox/Checkbox.test.tsx#L256